src/server.c

```diff
@@ -3974,6 +3971,11 @@ int processCommand(client *c) {
         if (listLength(server.ready_keys))
             handleClientsBlockedOnKeys();
     }
+
+    /* Update failed command calls if required */
+    if (previousErrCount < server.stat_total_error_replies) {
+        c->cmd->failed_calls++;
+    }
```

Comment:
this may need to be in `call()` not `processCommand()`.
let's try to think this though together.
when someone does:
```
MULTI
SET foo bar
LPUSH foo bla
EXEC
```
do we want the failure to be logged in LPUSH, EXEC, or both?
I think we only want it in LPUSH, so i think we need to move this block to `call`, but also add another global variable that will "consume" this diff in some way, so that EXEC will not show that error.
similar thing will happen for Lua and Modules doing RM_Call.

Reply:
@oranagra I've added two tests for multi and lua to showcase that with the latest push we're properly setting the error on the right command and not having it shown on exec/eval,etc... 
Tests: https://github.com/redis/redis/pull/8217/files#diff-595bcf8dbabe8679a083b9f9480389d8526888760af5de2a6941c715c2a00d82R23
I'm basically checking if we're within a multi context or lua or module client and if so I'm restoring the error back to the original previous value given the failed counter it has already been updated on the real command as follow:
```c
const long long prev_err_count = server.stat_total_error_replies;
    updateCachedTime(0);
    start = server.ustime;
    c->cmd->proc(c);
    duration = ustime()-start;
    dirty = server.dirty-dirty;
    if (dirty < 0) dirty = 0;

    /* Update failed command calls if required */
    if ((server.stat_total_error_replies - prev_err_count)>0) {
        real_cmd->failed_calls++;
        /* If we're within a MULTI context or LUA or RM_Call
         * we don't propagate the error uppwards */
        if ( c->flags & CLIENT_MULTI ||
             c->flags & CLIENT_LUA ||
             c->flags & CLIENT_MODULE ) {
            server.stat_total_error_replies=prev_err_count;
        }
    }
```
what do you guys think?

Reply:
@filipecosta90 i had something else in mind. 
Note that the way you implemented it, the errors inside the multi will not be shown in the global error counter. 

What I thought of doing is store another global variable (possibly a static inside the function), which recalls the last value of the global error counter which was used to increment the command error counter, so that one increment of the global error counter can't increment a command counter of more than once..

maybe it's even enough to make the stack variable you added in `call` a static one. Then, we update it in the entrance to `call`, and check again it at exit (before we update it again). 
This way we never set back the global error counter, but we also never log the same error twice in nested commands.

Reply:
@oranagra I've followed you recommendation with regards to the static variable within `call` and added an extra test (rejected call within multi/exec). 
Apart from that also checking the server stat total_error_replies value. 
Can you check it?